onnxruntime/core/providers/cpu/ml/tree_ensemble_helper.cc

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/cpu/ml/tree_ensemble_helper.h"
+#include "core/common/common.h"
+
+using namespace ::onnxruntime::common;
+using namespace std;
+namespace onnxruntime {
+namespace ml {
+
+template<typename TH>
+std::vector<TH> GetVectorAttrsOrDefault(const OpKernelInfo& info, const std::string& name, const std::vector<TH> default_value, 
+                                        ONNX_NAMESPACE::TensorProto_DataType proto_type) {
+  ONNX_NAMESPACE::TensorProto proto;
+  if (!info.GetAttr(name, &proto).IsOK()) {
+    return default_value;
+  }
+  auto n_dims = proto.dims_size();
+  if (n_dims == 0) {
+    return default_value;
+  }
+  ORT_ENFORCE(n_dims == 1, "Attribute '", name, "' must be a vector.");
+  ORT_ENFORCE(proto.data_type() == proto_type,
+              "Unexpected type (", proto.data_type(), "(for attribute '", name, "'.");
+  auto n_elements = proto.dims()[0];
+  ORT_ENFORCE(n_elements > 0, "Attribute '", name, "' has one dimension but is empty.");
+
+  if (proto_type == ONNX_NAMESPACE::TensorProto_DataType::TensorProto_DataType_DOUBLE) {
```

Comment:
>if (proto_type == ONNX_NAMESPACE::TensorProto_DataType::TensorProto_DataType_DOUBLE) { [](http://example.com/codeflow?start=2&length=86)

would it make sense to put this block early in the function? This check is not executed in certain cases.